java/java.source.base/src/org/netbeans/modules/java/source/parsing/FileObjects.java

```diff
@@ -949,6 +949,34 @@ static Pair<String,List<URL>> parseModulePatches(@NonNull final Iterator<? exten
         return null;
     }
 
+    public static URI getZipPathURI(URI zipURI, String resourceName) {
+        try {
+            //Optimistic try and see
+            return new URI ("jar:"+zipURI.toString()+"!/"+resourceName); //NOI18N
+        } catch (URISyntaxException e) {
+            //Need to encode the resName part (slower)
+            final StringBuilder sb = new StringBuilder ();
```

Comment:
how about?

 final StringBuilder sb = new StringBuilder (100).append("jar:").append(zipURI).append("!/");

and then just 

return new URI(sb.toString);